Rakefile

```diff
@@ -1,37 +1,11 @@
+# Load RSpec rake task as :spec
 require "rspec/core/rake_task"
 RSpec::Core::RakeTask.new(:spec)
 
-# The jasmine gem isn't designed for testing engines
-#
-# Proxy jasmine calls onto the dummy app's rakefile
-# Running jasmine from root prevents the spec files
-# from being compiled through the asset pipeline
-#
-# Import all dummy app rake tasks under a namespace
+# Load rake tasks from the dummy Rails app located at spec/dummy
+# This will also load rake tasks from lib/tasks
+APP_RAKEFILE = File.expand_path("spec/dummy/Rakefile", __dir__)
+load "rails/tasks/engine.rake"
```

Comment:
Nice this looks a lot cleaner.

As per your comment about not understanding where it came from, It runs this file: https://github.com/rails/rails/blob/main/railties/lib/rails/tasks/engine.rake which then executes loading the APP_RAKEFILE within a context of `app:` so all the tasks end up prefixed.

Reply:
Thanks for the explainer! 🤓 I'd tried and failed to find the source of `"rails/tasks/engine.rake"` – I hadn't realised it was in Railties.